### DISCUSSION TITLE
HTTP ERROR 403 No valid crumb was included in the request only in CURL and JAVA client

### DISCUSSION THREAD

**Discussion by sri1790** [zitadel/zitadel#5279] [Q&A] [answered]
I am trying to get token using the client credentials and getting success response in Postman and Go but same request is failing in Java and CURL. Please suggest.

**FAILURE**
<img width="988" alt="image" src="https://user-images.githubusercontent.com/23081940/220871821-24be5257-8223-44f2-a1ae-7eadacc12c7e.png">

<img width="646" alt="image" src="https://user-images.githubusercontent.com/23081940/220872087-3b8be322-d753-4bac-b49f-ffc6f2ab9313.png">

**SUCCESS**

<img width="571" alt="image" src="https://user-images.githubusercontent.com/23081940/220872207-43ab2de7-91c4-40e3-90a9-033af5ffd767.png">

<img width="623" alt="image" src="https://user-images.githubusercontent.com/23081940/220872538-e51740f4-e4d1-4cc6-9ebf-730d7bf22156.png">

**Comment by mffap** [comment]
Can you try without single quoting openid?

Here's an example curl:
https://zitadel.com/docs/apis/openidoauth/endpoints#client-credentials-grant

**Reply by sri1790** [reply]
@mffap ,

I tried and get the same error. but if you look in java http debug, everything is correct but getting the same error even with spring security

<img width="519" alt="image" src="https://user-images.githubusercontent.com/23081940/221180178-6d9d9817-6307-4feb-96f2-af35c7f1350b.png">

**Reply by mffap** [reply]
Seems odd that this message should come from zitadel. Do you have an existing session and being redirected to your app? What's the configured client app?

**Reply by fforootd** [reply]
hm that error does not look like from zitadel 😁

**Reply by sri1790** [reply]
this is M2M auth, so i tried with go, postman and wget are working but java  client (spring security, okhttp, apache http) and curl is failing. i debugged and checked all the headers and body all are matching and there is no difference. I couldn't able to find any issue on any sides. :( Please someone help me narrow down the issue. even i tried generating the code from postman request and that also failing.

**Reply by fforootd** [reply]
Can you share a little more about your setup? What does java do?

Is it a client that calls the token endpoint, or is it passing along the token to zitadel?

**Comment by sri1790** [comment, marked as answer]
i finally found the issue, seems two applications listing in single port. 
Apologies for inconvince. No issue at ZITADEL

**Reply by fforootd** [reply]
Great, let us know if something does not work out!
